### ISSUE TITLE
Rule request: default parameters at the end

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#2176]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

From [Swift's API Design Guidelines](https://swift.org/documentation/api-design-guidelines/):

> Prefer to locate parameters with defaults toward the end of the parameter list. Parameters without defaults are usually more essential to the semantics of a method, and provide a stable initial pattern of use where methods are invoked.

Thanks @mokagio for pointing this potential rule [on Twitter](https://twitter.com/mokagio/status/991626605817548801).

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// would trigger
func foo(bar: Int = 0, baz: String) {}

// wouldn't trigger
func foo(baz: String, bar: Int = 0) {}
```

3. Should the rule be configurable, if so what parameters should be configurable?

Just severity. 

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.

Probably enabled by default, depends on how many violation oss-check finds.

**Comment by SDGGiesbrecht** [comment]
Could the rule be designed to work better with trailing closure syntax?

i.e. “Default parameters should come before closure parameters, but after any other kind.”

That is probably the reason for the softer wording in the API guidelines: “*Prefer* to locate parameters [...] *toward* the end [...]”.

**Comment by bull-xu-stride** [comment]
There is an issue on this rule:

```
typealias AttributedFormat = ((_ attr: NSMutableAttributedString) -> Void)

extension String {

	func attributed(_ format: AttributedFormat = { _ in }) -> NSMutableAttributedString {

		let attr = NSMutableAttributedString(string: self)
		format(attr)
		return attr

	}

}
```
